### PR TITLE
[Java] Add per job runtime env env vars.

### PR DESCRIPTION
**Pull request by jovany-wang**
## Why are these changes needed?
1. Support setting environment variables in runtime env for a job, like:
```yaml
ray : {
  job : {
    runtime-env: {
         // Environment variables to be set on worker processes in current job.
         "env-vars": {
           // key1: "value11"
           // key2: "value22"
         }
      }
   }
}
```
It could be set by system properties before `Ray.init()` as well:
```java
  System.setProperty("ray.job.runtime-env.env-vars.KEY1", "A");
  System.setProperty("ray.job.runtime-env.env-vars.KEY2", "B");

  Ray.init();
```

2. Setting environment variables for an actor will overwrite and merge to the environment variables of job.
```java
System.setProperty("ray.job.runtime-env.env-vars.KEY1", "A");
System.setProperty("ray.job.runtime-env.env-vars.KEY2", "B");

Ray.init();
  
RuntimeEnv runtimeEnv = new RuntimeEnv.Builder().addEnvVar("KEY1", "C").build();

/// actor1 has the env vars: {"KEY1" : "C", "KEY2" : "B"}
ActorHandle<A> actor1 = Ray.actor(A::new).setRuntimeEnv(runtimeEnv).remote();
/// actor2 has the env vars: {"KEY1" : "A", "KEY2" : "B"} 
ActorHandle<A> actor2 = Ray.actor(A::new).remote();
```

## Related issue number
#20425

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jovany-wang** [comment]
Waiting for the CI result.

File: java/runtime/src/main/resources/ray.default.conf [hunk 1]
```diff
@@ -31,11 +31,20 @@ ray {
...
+    runtime-env: {
+         // Environment variables to be set on worker processes in current job.
+         "env-vars": {
+           // key1: "value11"
+           // key2: "value22"
+         }
+
+         "jars":[
+             // Job agent will download these jars, add them to the classpath.
+             "https://host/a.jar"
```
Comment (SongGuyang) [inline java/runtime/src/main/resources/ray.default.conf:44]: Is this conf file visible for users? If it is, can we remove this `jars` from current PR? If not, can we add a comment here to notice that this `jars` API is not implemented yet?

**Review by SongGuyang — APPROVED** [review]
Looks good overall. Left two comments.

File: java/runtime/src/main/resources/ray.default.conf [hunk 2]
```diff
@@ -31,11 +31,20 @@ ray {
     num-java-workers-per-process: 1
     /// The jvm options for java workers of the job.
     jvm-options: []
-    // Environment variables to be set on worker processes.
-    worker-env {
-      // key1 : "value1"
-      // key2 : "value2"
+
+    runtime-env: {
```
Comment (SongGuyang) [inline java/runtime/src/main/resources/ray.default.conf:35]: Can we move `runtime-env` out from `job` field? It is not in job config in Python API.

File: java/runtime/src/main/resources/ray.default.conf (hunk 1 above)
Comment (jovany-wang) [reply java/runtime/src/main/resources/ray.default.conf:44]: I'm going to remove it.

File: java/runtime/src/main/resources/ray.default.conf (hunk 2 above)
Comment (jovany-wang) [reply java/runtime/src/main/resources/ray.default.conf:35]: Runtime env is set for job.  This is the way we align it to job config in python.

File: java/runtime/src/main/java/io/ray/runtime/config/RayConfig.java [hunk 3]
```diff
@@ -54,6 +53,8 @@
 
   public int runtimeEnvHash;
 
+  public RuntimeEnvImpl runtimeEnvImpl = null;
```
[the remark this answers is not in the document]
Comment (jovany-wang) [reply java/runtime/src/main/java/io/ray/runtime/config/RayConfig.java:56]: It's in runtime module and that's what we use internally. No need to use interface.

[the remark this answers is not in the document]
Comment (SongGuyang) [reply java/runtime/src/main/java/io/ray/runtime/config/RayConfig.java:56]: A little weird, why we don't use interface when we already have? `RayConfig` don't need to know implementation of `RuntimeEnv`. And we also need a `getEnvVars` method in the `RuntimeEnv` interface.

File: java/runtime/src/main/resources/ray.default.conf (hunk 2 above)
Comment (jovany-wang) [reply java/runtime/src/main/resources/ray.default.conf:35]: If you have any concern on this structure, you could file an issue for that, and then do more discussions. This PR is not aiming to change the original structure.

**Pull request merged**